test/NuGet.Core.Tests/NuGet.Protocol.Tests/Plugins/PluginDiscovererTests.cs

```diff
@@ -20,15 +23,21 @@ public class PluginDiscovererTests
         [InlineData(" ")]
         public void Constructor_AcceptsAnyString(string rawPluginPaths)
         {
-            using (new PluginDiscoverer(rawPluginPaths))
+            var environmentalVariableReader = new Mock<IEnvironmentVariableReader>();
+            environmentalVariableReader.Setup(env => env.GetEnvironmentVariable(EnvironmentVariableConstants.DesktopPluginPaths)).Returns(rawPluginPaths);
+            environmentalVariableReader.Setup(env => env.GetEnvironmentVariable(EnvironmentVariableConstants.CorePluginPaths)).Returns(rawPluginPaths);
+            using (new PluginDiscoverer())
```

Comment:
Copilot suggested a better way to check if constructor raises an exception or not. The current implementation throws an exception and causes the entire test suite to stop execution.

```cs
Exception exception = Record.Exception(() =>
    {
        using (new PluginDiscoverer())
        {
        }
    });

    Assert.Null(exception); // Assert that no exception was throw
```